app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/vm/operations.rb

```diff
@@ -15,4 +15,30 @@ def raw_create_snapshot(name, desc = nil, _memory = nil)
     create_notification(:vm_snapshot_failure, :error => err.to_s, :snapshot_op => "create")
     raise MiqException::MiqVmSnapshotError, err.to_s
   end
+
+  def raw_remove_snapshot(snapshot_id)
+    with_provider_connection(:service => 'PCloudSnapshotsApi') do |api|
+      snapshot = Snapshot.find_by(:id => snapshot_id)
```

Comment:
So `Model.find_by(:id => id)` will return `nil` if the record cannot be found where `Model.find(id)` will raise a RecordNotFound exception but will guarantee that the object you get back is not `nil`.

Issue being if someone deleted the snapshot from the database after this message was queued but before this is executed then `snapshot` would be `nil` here and the line below would raise an exception.

Recommend either `snapshot = Snapshot.find(snapshot_id)` or keep what you current have but check if `snapshot` is `nil` and handle that condition

Reply:
Great points. Thanks for the suggested code change. Tested fine and included in the newest push.